taichi/backends/vulkan/runtime.cpp

```diff
@@ -411,26 +408,46 @@ VkRuntime::KernelHandle VkRuntime::register_taichi_kernel(
 
 void VkRuntime::launch_kernel(KernelHandle handle, RuntimeContext *host_ctx) {
   auto *ti_kernel = ti_kernels_[handle.id_].get();
+
+  std::unique_ptr<DeviceAllocationGuard> ctx_buffer_host, ctx_buffer;
+
+  if (ti_kernel->get_ctx_buffer_size()) {
+    ctx_buffer = device_->allocate_memory_unique(
```

Comment:
Would this result in memory fragmentation? Or the VMA should take care of it?

Reply:
It would, let's do the naive one for now until we kick ext arr out of context buffer and add a ring buffer

Reply:
SG